code/go/0chain.net/blobbercore/allocation/attributesfilechange.go

```diff
@@ -38,13 +39,14 @@ func (ac *AttributesChange) ApplyChange(ctx context.Context, _ *AllocationChange
 		dirRef    = ref
 		treelevel = 0
 	)
-
+	dirRef.HashToBeComputed = true
```

Comment:
What's rule for setting this `HashToBeComputed`, i.e when should we set it to true, and when to false

Reply:
For the HashToBeComputed flag parameter is the result of this: https://github.com/0chain/blobber/pull/529#discussion_r816455077 
since whenever there's new/update/delete file we have to update the hash but we can use hashes that did not change to optimise the calculations.